app-typescript/components/ToastMessage.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import ToastText from './ToastText';
+import classNames from 'classnames';
+
+export type MessageProp =
+    | React.ReactNode
+    | string;
+
+export type Position = 'top' | 'bottom' | 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left';
+
+export type NotesType = 'default' | 'primary' | 'success' | 'warning' | 'alert' | 'highlight' | 'light';
+
+export interface IMessageOptions {
+    message?: MessageProp;
+    id?: string;
+    duration?: number | null;
+    type?: NotesType;
+    position?: Position;
+    icon?: string;
+    size?: 'fixed-s' | 'fixed-m' | 'fixed-l' | 'fixed-xl';
+}
+
+interface IProps extends IMessageOptions {
+    closeElement(id: string, position: Position): void;
+}
+
+export const ToastMessage = ({
+    id,
+    message,
+    type,
+    icon,
+    size,
+    duration,
+    position,
+    closeElement,
+}: IProps) => {
+    const [show, setShow] = React.useState();
+    const [enter, setEnter] = React.useState(false);
+    let timer = null;
+
+    React.useEffect(() => {
+        if (show === undefined) {
+            setShow(true);
```

Comment:
I think this should be set via `React.useState(true);`, single `useEffect` should handle single state bit and not more imo